crates/sui-rosetta/src/account.rs

```diff
@@ -29,108 +31,97 @@ pub async fn balance(
 ) -> Result<AccountBalanceResponse, Error> {
     env.check_network_identifier(&request.network_identifier)?;
     let address = request.account_identifier.address;
+    let currencies = &request.currencies;
     let mut retry_attempts = 5;
-    if let Some(SubAccount { account_type }) = request.account_identifier.sub_account {
-        while retry_attempts > 0 {
-            let balances_first =
-                get_sub_account_balances(account_type.clone(), &ctx.client, address).await?;
-            let checkpoint1 = ctx
-                .client
-                .read_api()
-                .get_latest_checkpoint_sequence_number()
-                .await?;
-            // Get another checkpoint which is greater than current
-            let mut checkpoint2 = ctx
-                .client
-                .read_api()
-                .get_latest_checkpoint_sequence_number()
-                .await?;
-
-            while checkpoint2 <= checkpoint1 {
-                checkpoint2 = ctx
-                    .client
-                    .read_api()
-                    .get_latest_checkpoint_sequence_number()
-                    .await?;
-                tokio::time::sleep(Duration::from_secs(1)).await;
-            }
-            let balances_second =
-                get_sub_account_balances(account_type.clone(), &ctx.client, address).await?;
-            if balances_first.eq(&balances_second) {
-                return Ok(AccountBalanceResponse {
-                    block_identifier: ctx.blocks().create_block_identifier(checkpoint2).await?,
-                    balances: balances_first,
-                });
-            } else {
-                // retry logic needs to be aaded
-                retry_attempts -= 1;
-            }
+    while retry_attempts > 0 {
+        let balances_first = get_balances(&ctx, &request, address, currencies.clone()).await?;
+        let checkpoint1 = get_checkpoint(&ctx).await.unwrap();
+        let mut checkpoint2 = get_checkpoint(&ctx).await.unwrap();
```

Comment:
These unwraps are scary. I think they should all be `.await?` instead,
or a clear explanation why this unwrap cannot panic.